tests/attr/test_saliency.py

```diff
@@ -109,11 +112,12 @@ def _saliency_classification_assert(self, nt_type="vanilla"):
 
             output = model(input)[:, target]
             output.backward()
-            expected = torch.abs(input.grad)
-            self.assertEqual(
-                expected.detach().numpy().tolist(),
-                attributions.detach().numpy().tolist(),
-            )
+            if input.grad:
```

Comment:
`input.grad` is typed as Optional[Tensor], and is shouldn't be None by any chance. Thus I add a if check here to make mypy happy